pkg/zkproofs/range.go

```diff
@@ -0,0 +1,144 @@
+package zkproofs
+
+import (
+	crand "crypto/rand"
+	"encoding/json"
+	"errors"
+	"github.com/coinbase/kryptology/pkg/bulletproof"
+	"github.com/coinbase/kryptology/pkg/core/curves"
+	"github.com/gtank/merlin"
+	"github.com/sei-protocol/sei-cryptography/pkg/encryption/elgamal"
+)
+
+const rangeDomain = "rangeDomain"
+const ippDomain = "ippDomain"
+
+type RangeProof struct {
+	Proof      *bulletproof.RangeProof
+	Randomness curves.Point
+	UpperBound int
+}
+
+// NewRangeProof a range proof for some ciphertext that proves the value is between 0 and 2^upperBound
+// Note that while this proof generator takes in the plaintext of the value, it will only work in conjunction with a
+// ciphertext that encrypts value using the given randomness.
```

Comment:
// Params:
// upperBound: The upper bound of the range we want prove the value lies within, calculated as 2^upperBound
// value: The value encrypted by the ciphertext on which we are creating this proof for
// randomness: The randomness used in the generation of the ciphertext on which we are creating this proof for